test/e2e/route_advertisements.go

```diff
@@ -178,37 +180,217 @@ var _ = ginkgo.Describe("BGP: Pod to external server when default podNetwork is
 				}
 			}
 
-			ginkgo.By("queries to the external server are not SNATed (uses podIP)")
+			// Helper function to test pod to server connectivity and verify source IP
+			testClientPodToServerConnectivity := func(serverIPs []string, portNumber string, expectedIPs []string, serverDesc string) bool {
```

Comment:
Can you define this helper function out of the test? So it can be reused by others.

Reply:
moved to util.go now, thanks!